ddtrace/tracer/tracer.go

```diff
@@ -702,7 +703,12 @@ func startExecutionTracerTask(ctx gocontext.Context, span *span) (gocontext.Cont
 		// skipped.
 		ctx = globalinternal.WithExecutionNotTraced(ctx)
 	}
-	rt.Log(ctx, "span id", strconv.FormatUint(span.SpanID, 10))
+	var b [8]byte
+	binary.LittleEndian.PutUint64(b[:], span.SpanID)
+	// TODO: can we make string(b[:]) not allocate? e.g. with unsafe
+	// shenanigans? rt.Log won't retain the message string, though perhaps
+	// we can't assume that will always be the case.
+	rt.Log(ctx, "datadog.uint64_span_id", string(b[:]))
```

Comment:
From Go 1.20 we can use [`unsafe.String`](https://pkg.go.dev/unsafe@master#String) but looking at the repository and from what I heard, I think that we don't want `unsafe` calls in the tracer.

Reply:
@darccio 👍 

Let's not worry about it as part of this PR. The previous code already caused an allocation as well as far as I can tell. We can revisit this for flight recording mode.